central/config/datastore/datastore.go

```diff
@@ -50,6 +51,60 @@ func (d *datastoreImpl) UpsertConfig(ctx context.Context, config *storage.Config
 	} else if !ok {
 		return sac.ErrResourceAccessDenied
 	}
+	if privateConf := config.GetPrivateConfig(); privateConf != nil {
+		if clusterRetentionConf := privateConf.GetDecommissionedClusterRetention(); clusterRetentionConf != nil {
+			hasUpdate, err := d.hasClusterRetentionConfigUpdate(ctx, clusterRetentionConf)
+			if err != nil {
+				return err
+			}
 
+			if hasUpdate {
+				clusterRetentionConf.LastUpdated = types.TimestampNow()
+			}
+		}
+	}
 	return d.store.Upsert(ctx, config)
 }
+
+func (d *datastoreImpl) hasClusterRetentionConfigUpdate(ctx context.Context,
+	newConf *storage.DecommissionedClusterRetentionConfig) (bool, error) {
+	conf, err := d.getClusterRetentionConfig(ctx)
+	if err != nil {
+		return false, err
+	}
+	return !clusterRetentionConfigsEqual(conf, newConf), nil
+}
+
+func (d *datastoreImpl) getClusterRetentionConfig(ctx context.Context) (*storage.DecommissionedClusterRetentionConfig, error) {
+	conf, err := d.GetConfig(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if privateConf := conf.GetPrivateConfig(); privateConf != nil {
+		return privateConf.GetDecommissionedClusterRetention(), nil
+	}
+	return nil, nil
```

Comment:
Getters handle nil check. 
nit: `return conf.GetPrivateConfig().GetDecommissionedClusterRetention(), nil`

Reply:
That would have 💣 if `GetPrivateConfig` can return nil but it cannot be:
https://github.com/stackrox/stackrox/blob/8864b67f8b7f5c64601dcdabb88cd33e122ee9e0/central/config/datastore/singleton.go#L77

Hence the chaining will work.